react-components/src/components/RuleBasedOutputs/utils.ts

```diff
@@ -0,0 +1,218 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import { Color } from 'three';
+import {
+  type StringExpression,
+  type ColorRuleOutput,
+  type NodeAndRange,
+  type RuleOutput,
+  type NumericExpression,
+  type MetadataRuleTrigger,
+  type Expression,
+  type RuleOutputSet,
+  type Rule
+} from './types';
+import { type CogniteCadModel, TreeIndexNodeCollection } from '@cognite/reveal';
+import { type AssetMapping3D, type Asset } from '@cognite/sdk';
+
+const checkStringExpressionStatement = (asset: Asset, expression: StringExpression): boolean => {
+  const { trigger, condition } = expression;
+
+  let expressionResult: boolean = false;
+
+  switch (condition.type) {
+    case 'equals': {
+      expressionResult = asset[trigger.type]?.[trigger.key] === condition.parameter;
+      break;
+    }
+    case 'notEquals': {
+      expressionResult = asset[trigger.type]?.[trigger.key] !== condition.parameter;
+      break;
+    }
+    case 'contains': {
+      expressionResult = asset[trigger.type]?.[trigger.key].includes(condition.parameter) ?? false;
```

Comment:
Hmm yeah good point. Maybe we can send out a undefined (instead of false) and if the not-clause receives that then it keeps false. But I think we can test it more and see how it goes, or do you prefer that I change it now and see if I have room for testing before merging? I believe we can do it during the next comings more carefully and keep it like this for now, what do you think?

Reply:
Did some tweaks based on my thoughts in here:
https://github.com/cognitedata/reveal/pull/4210/commits/e4c1637a2c6f4412f8fa79861d3d37844a646860

Reply:
Looks good! :+1: